### PR TITLE
feat(webpack-3): upgrade to webpack 3

### PR DESCRIPTION
**Pull request by sbolel**
### Summary of Changes

- upgrade to webpack 3
- add copying minified js to example folder in build
- add npm-run-all to run parallel tasks
- yarn.lock versions

### Demo

https://sbolel.github.io/react-google-button/

### CODE REVIEW

File: examples/basic/index.html [hunk 1]
```diff
@@ -8,7 +8,7 @@
   <script src="https://unpkg.com/react-dom@15.3.1/dist/react-dom.js"></script>
   <script src="https://unpkg.com/prop-types/prop-types.min.js"></script>
   <!-- For local development remove the following line: -->
-  <script src="https://unpkg.com/react-google-button@latest/dist/react-google-button.js"></script>
+  <script src="react-google-button.js"></script>
```
Comment (prescottprue) [inline examples/basic/index.html]: How does this work? Is it supposed to point to the one in the dist folder? It won't really work as a standalone demo then right?

File: package.json [hunk 2]
```diff
@@ -86,6 +86,6 @@
     "react-addons-test-utils": "^15.5.1",
     "react-dom": "15.4.2",
     "rimraf": "^2.5.1",
-    "webpack": "^2.6.0"
+    "webpack": "^3.0.0"
```
Comment (prescottprue) [inline package.json]: Nice!

**Review by prescottprue — COMMENTED** [review]
Just a question about the demo stuff

File: examples/basic/index.html (hunk 1 above)
Comment (prescottprue) [reply examples/basic/index.html]: Ahh.. I see, you copied the file in below. We should maybe do that as part of the build step?

Comment (sbolel) [reply examples/basic/index.html]: I'm on board with that plan!

File: package.json (hunk 2 above)
Comment (sbolel) [reply package.json]: Pretty cool! I may have done a double take when it just... Worked. Haha.

File: examples/basic/index.html (hunk 1 above)
Comment (sbolel) [reply examples/basic/index.html]: If you'd like, I can add that to this PR and open it against your repo 😁

Comment (sbolel) [reply examples/basic/index.html]: ...and remove the file I copied.

Comment (prescottprue) [reply examples/basic/index.html]: 💯

File: package.json (hunk 2 above)
Comment (sbolel) [reply package.json]: I'll get this in there this weekend @prescottprue :)

Comment (prescottprue) [reply package.json]: They are already up to `3.4.1` 😆

**Pull request closed without merging**